snuba/subscriptions/scheduler_processing_strategy.py

```diff
@@ -257,3 +278,167 @@ def terminate(self) -> None:
+        scheduled_topic_spec: KafkaTopicSpec,
+        commit: Callable[[Mapping[Partition, Position]], None],
+    ) -> None:
+        self.__schedulers = schedulers
+        self.__encoder = SubscriptionScheduledTaskEncoder(entity_key)
+        self.__producer = producer
+        self.__scheduled_topic = Topic(scheduled_topic_spec.topic_name)
+        self.__commit = commit
+        self.__closed = False
+
+        # Stores each tick with it's futures
+        self.__queue = ScheduledSubscriptionQueue()
+
+        # Not a hard max
+        self.__max_buffer_size = 10000
+
+    def poll(self) -> None:
+        # Remove completed tasks from the queue and raise if an exception occurred.
+        # This method does not attempt to recover from any exception.
+        # Also commits any offsets required.
+        while True:
+            tick_subscription = self.__queue.peek()
+
+            if tick_subscription is None:
+                break
+
+            if not tick_subscription.subscription_future.done():
+                break
+
+            tick_subscription.subscription_future.result()
+
+            self.__queue.popleft()
+
+            if tick_subscription.should_commit:
+                self.__commit(
+                    {
+                        tick_subscription.tick_message.partition: Position(
+                            tick_subscription.tick_message.offset,
+                            tick_subscription.tick_message.timestamp,
+                        )
+                    }
+                )
+
+    def submit(self, message: Message[CommittableTick]) -> None:
+        assert not self.__closed
+
+        # If queue is full, raise MessageRejected to tell the stream
+        # processor to pause consuming
+        if len(self.__queue) >= self.__max_buffer_size:
+            raise MessageRejected
+
+        # Otherwise, add the tick message and all of it's subscriptions to
+        # the queue
+        tick = message.payload.tick
+        tasks = self.__schedulers[tick.partition].find(tick.timestamps)
+        self.__queue.append(
+            message,
+            deque(
+                [
+                    self.__producer.produce(
+                        self.__scheduled_topic, self.__encoder.encode(task)
+                    )
+                    for task in tasks
+                ]
+            ),
+        )
+
+    def close(self) -> None:
+        self.__closed = True
+
+    def terminate(self) -> None:
+        self.__closed = True
+
+    def join(self, timeout: Optional[float] = None) -> None:
+        start = time.time()
+
+        while self.__queue:
+            remaining = timeout - (time.time() - start) if timeout is not None else None
+
+            if remaining is not None and remaining <= 0:
+                logger.warning(f"Timed out with {len(self.__queue)} futures in queue")
+                break
+
+            self.__queue.popleft().subscription_future.result(remaining)
```

Comment:
You are not committing here. I think you should still be committing all  ticks that have `should_commit` set to true

Reply:
Thanks, added committing here to join as well